lib/gvl-tracing.rb

```diff
@@ -28,6 +28,53 @@
 require_relative "gvl_tracing/version"
 
 require "gvl_tracing_native_extension"
+require "json"
 
 module GvlTracing
+  @path = "/tmp/gvl-tracing.json"
+
+  class << self
+    private :_start
+    private :_stop
+
+    def start(file)
+      @path = file
+      _start(@path)
+    end
```

Comment:
There's a bit of validation logic in `_start` which may actually make it not start, thus the `@path` should only be updated after `_start` returns without an exception:

```suggestion
    def start(file)
      _start(file)
      @path = file
    end
```

In particular, the following issue could happen before:

```ruby
GvlTracing.start("foo.json")
GvlTracing.start("bar.json") rescue nil # This fails, but the `@path` gets changed
GvlTracing.stop # The events are in `foo.json`, the thread names end up in `bar.json`
```